srtcore/core.cpp

```diff
@@ -10360,6 +10344,24 @@ int srt::CUDT::processData(CUnit* in_unit)
             }
         }
 
+        if (!srt_loss_seqs.empty())
+        {
+            if (initial_loss_ttl)
+            {
+                // pack loss list for (possibly belated) NAK
+                // The LOSSREPORT will be sent in a while.
+                ScopedLock lock(m_RcvLossLock);
+                for (loss_seqs_t::iterator i = srt_loss_seqs.begin(); i != srt_loss_seqs.end(); ++i)
+                {
+                    m_FreshLoss.push_back(CRcvFreshLoss(i->first, i->second, initial_loss_ttl));
+                }
+                HLOGC(qrlog.Debug,
+                      log << CONID() << "FreshLoss: added sequences: " << Printable(srt_loss_seqs)
+                          << " tolerance: " << initial_loss_ttl);
+                reorder_prevent_lossreport = true;
+            }
+        }
```

Comment:
I think it can be even combined with 

```c++
    if (!srt_loss_seqs.empty())
    {
        // A loss is detected
        {
            // TODO: Can unlock rcvloss after m_pRcvLossList->insert(...)?
            // And probably protect m_FreshLoss as well.

            HLOGC(qrlog.Debug, log << "processData: LOSS DETECTED, %: " << Printable(srt_loss_seqs) << " - RECORDING.");
            // if record_loss == false, nothing will be contained here
            // Insert lost sequence numbers to the receiver loss list
            ScopedLock lg(m_RcvLossLock);
            for (loss_seqs_t::iterator i = srt_loss_seqs.begin(); i != srt_loss_seqs.end(); ++i)
            {
                // If loss found, insert them to the receiver loss list
                m_pRcvLossList->insert(i->first, i->second);
            }
        }
```

Reply:
You are doing it in the next commit. Missed that 🙂

Reply:
Yes, in case we need to revert something.